### PR TITLE
Feature/job ad building

### PR DESCRIPTION
**Pull request by developstuff93**
I want to merge this change because...

### Pull Request Checklist

1. [ ] All visible strings are translated with proper context.
1. [ ] All data-formatting is locale-aware (dates, numbers, and so on).
1. [ ] The changes are tested.
1. [ ] The code is documented (docstrings, project documentation).
1. [ ] Changes are mentioned in the changelog.

### Test Environment Config

API_URI=https://qa.staging.saleor.cloud/graphql/

### CODE REVIEW

File: src/@next/CheckoutUtils/constants.tsx [hunk 1]
```diff
@@ -1,4 +1,4 @@
-export const CheckoutMetadataTypes = {
+export const CheckoutMetadataTypes: any = {
```
Comment (E1379) [inline src/@next/CheckoutUtils/constants.tsx]: Making it any, makes it imposible for TS to validate. I think `CheckoutMetadataTypes` and `CheckoutValueTypes` should implemented as enum: https://www.typescriptlang.org/docs/handbook/enums.html#string-enums

File: src/@next/pages/CheckoutPage/CheckoutOrder.tsx [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+const CheckoutOrder = () => {
+  const { loaded: cartLoaded, items } = useCart();
+
+  if (!cartLoaded) {
+    return <Loader />;
+  }
+
+  if (!items || !items.length) {
+    return <div>No products in the basket</div>;
```
Comment (E1379) [inline src/@next/pages/CheckoutPage/CheckoutOrder.tsx]: Make this be translatable

File: src/@next/pages/CheckoutPage/CheckoutPage copy.tsx [hunk 3]
```diff
@@ -0,0 +1,398 @@
...
+    if (
+      !submitInProgress &&
+      checkout &&
+      pathname === paymentConfirmStepLink &&
+      !paymentConfirmation
+    ) {
+      handlePaymentConfirm();
+    }
+  }, [pathname, query, submitInProgress, checkout]);
+  // console.log(activeStep.step);
```
Comment (E1379) [inline src/@next/pages/CheckoutPage/CheckoutPage copy.tsx]: Should be removed

File: src/@next/pages/CheckoutPage/CheckoutPage copy.tsx [hunk 4]
```diff
@@ -0,0 +1,398 @@
...
+          "data_checkout",
+          JSON.stringify({
+            ...newCheckout,
+            lines: items,
+            metadata: {},
+          })
+        );
+        history.go(0);
+      } catch (error) {
+        console.log("Error on creating checkout", error); // eslint-disable-line no-console
```
Comment (E1379) [inline src/@next/pages/CheckoutPage/CheckoutPage copy.tsx:344]: Must the user not be informed as well that something went wrong? Maybe new ticket?

File: src/@next/pages/CheckoutPage/CheckoutPage.tsx [hunk 5]
```diff
@@ -1,398 +1,55 @@
...
-      if (paymentStepLink) {
-        push(paymentStepLink);
-        setSubmitInProgress(false);
-        setPaymentConfirmation(false);
-      }
+      });
+      const newCheckoutToken = response?.data?.checkoutCreate?.checkout?.token;
+      replace(`${paths.checkoutCreateJobAd}/?id=${newCheckoutToken}`);
+    } catch (error) {
+      console.log("Error on creating checkout", error); // eslint-disable-line no-console
```
Comment (E1379) [inline src/@next/pages/CheckoutPage/CheckoutPage.tsx:43]: Must the user not be informed as well that something went wrong? Maybe new ticket?

File: src/@next/pages/CreateJobAd/CreateJobAdContent.tsx [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+  setMetaFieldData: Function;
+}) => {
+  const classes = useStyles();
+  return (
+    <Container maxWidth="xl" className={classes.root}>
+      <Grid container justify="center">
+        <Grid item xs={12}>
+          <Box className={classes.header}>
+            <Typography variant="h4" component="h1">
+              Create your job ad(s)
```
Comment (E1379) [inline src/@next/pages/CreateJobAd/CreateJobAdContent.tsx]: Make it translatable

File: src/@next/pages/CreateJobAd/CreateJobAdContent.tsx [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+            </Typography>
+            <Typography variant="body1" component="h6">
+              Fill in the needed information for your job posting(s). Some
```
Comment (E1379) [inline src/@next/pages/CreateJobAd/CreateJobAdContent.tsx]: Make it translatable

File: src/@next/pages/CreateJobAd/JobCriteriaForm.tsx [hunk 8]
```diff
@@ -0,0 +1,270 @@
...
+}: {
+  metaErrors: any;
+  metadataValues: any;
+  setMetaFieldData: Function;
+}) => {
+  const classes = useStyles();
+
+  return (
+    <JobAdStepContainer>
+      <JobStepNumber number={2} title="Job criteria" />
```
Comment (E1379) [inline src/@next/pages/CreateJobAd/JobCriteriaForm.tsx]: make it translatable.

File: src/@next/pages/CreateJobAd/JobCriteriaForm.tsx [hunk 9]
```diff
@@ -0,0 +1,270 @@
...
+      {/* Min Education Level */}
+      <JobPostingFieldContainer title="Minimum level of education">
```
Comment (E1379) [inline src/@next/pages/CreateJobAd/JobCriteriaForm.tsx]: make it translatable.

File: src/@next/pages/CreateJobAd/JobPostingDetailsForm.tsx [hunk 10]
```diff
@@ -0,0 +1,171 @@
...
+  metaErrors: any;
+  setMetaFieldData: Function;
+}) => {
+  return (
+    <JobAdStepContainer>
+      <JobStepNumber number={1} title="Job posting details" />
+      {/* Job Title */}
+      <JobPostingFieldContainer
+        title="Job title"
+        description="This is the job title which will be shown to your candidates:"
```
Comment (E1379) [inline src/@next/pages/CreateJobAd/JobPostingDetailsForm.tsx]: make all text literals in PR translatable.

**Review by E1379 — CHANGES_REQUESTED** [review]

File: src/@next/CheckoutUtils/constants.tsx (hunk 1 above)
Comment (developstuff93) [reply src/@next/CheckoutUtils/constants.tsx]: Updated as enum

File: src/@next/pages/CheckoutPage/CheckoutOrder.tsx (hunk 2 above)
Comment (developstuff93) [reply src/@next/pages/CheckoutPage/CheckoutOrder.tsx]: Done

File: src/@next/pages/CheckoutPage/CheckoutPage copy.tsx (hunk 3 above)
Comment (developstuff93) [reply src/@next/pages/CheckoutPage/CheckoutPage copy.tsx]: Removed

File: src/@next/pages/CheckoutPage/CheckoutPage copy.tsx (hunk 4 above)
Comment (developstuff93) [reply src/@next/pages/CheckoutPage/CheckoutPage copy.tsx:344]: Maybe, not sure how can we display this kind of errors for now

Comment (developstuff93) [reply src/@next/pages/CheckoutPage/CheckoutPage copy.tsx:344]: `CheckoutPage copy.tsx` we don't use this anywhere, Just made a copy for the reference

File: src/@next/pages/CheckoutPage/CheckoutPage.tsx (hunk 5 above)
Comment (developstuff93) [reply src/@next/pages/CheckoutPage/CheckoutPage.tsx:43]: > not sure how can we display this kind of errors for now

File: src/@next/pages/CreateJobAd/CreateJobAdContent.tsx (hunk 6 above)
Comment (developstuff93) [reply src/@next/pages/CreateJobAd/CreateJobAdContent.tsx]: Done

File: src/@next/pages/CreateJobAd/CreateJobAdContent.tsx (hunk 7 above)
Comment (developstuff93) [reply src/@next/pages/CreateJobAd/CreateJobAdContent.tsx]: Updated it

File: src/@next/pages/CreateJobAd/JobCriteriaForm.tsx (hunk 8 above)
Comment (developstuff93) [reply src/@next/pages/CreateJobAd/JobCriteriaForm.tsx]: Done

File: src/@next/pages/CreateJobAd/JobCriteriaForm.tsx (hunk 9 above)
Comment (developstuff93) [reply src/@next/pages/CreateJobAd/JobCriteriaForm.tsx]: It's now translatable

File: src/@next/pages/CreateJobAd/JobPostingDetailsForm.tsx (hunk 10 above)
Comment (developstuff93) [reply src/@next/pages/CreateJobAd/JobPostingDetailsForm.tsx]: I tried to update all, should be good for now.

**Comment by developstuff93** [comment]
@E1379 I've changed all the text to be able to translated. Can you please take a look again?

**Pull request closed without merging**